crates/nostr-relay-builder/src/local/inner.rs

```diff
@@ -316,6 +316,36 @@ impl InnerLocalRelay {
                         .await;
                 }
 
+                if !event.verify_id() {
+                    return send_msg(
+                            ws_tx,
+                            RelayMessage::Ok {
+                                event_id: event.id,
+                                status: false,
+                                message: Cow::Owned(format!(
+                                    "{}: invalid event ID",
+                                    MachineReadablePrefix::Invalid
+                                )),
+                            },
+                        )
+                        .await;
+                }
+
```

Comment:
I think could make sense to check before the POW and later the signature. On my machine, the POW check takes ~0.2ns, while the event signature ~30000ns, so if it's set a min POW, the relay avoids always verifying the signature, which is heavier.

So:
- check ID
- check POW
- check signature

What do you think?

```bash
test event::benches::event_check_pow                      ... bench:           0.22 ns/iter (+/- 0.00)
test event::benches::verify_event_id                      ... bench:         786.87 ns/iter (+/- 27.87)
test event::benches::verify_event_sig                     ... bench:      31,998.58 ns/iter (+/- 729.32)
```